### PR TITLE
Adjusting hf_msdsal standalone module

### PR DESCRIPTION
**Pull request by penegui**
Adjusting hf_msdsal standalone module

### CODE REVIEW

File: armsrc/iso14443a.c [hunk 1]
```diff
@@ -1183,7 +1188,15 @@ bool SimulateIso14443aInit(uint8_t tagType, uint16_t flags, uint8_t *data, tag_r
 
         // Configure the ATQA and SAK accordingly
         rATQA[0] &= 0xBF;
-        rSAKc1[0] = sak & 0xFB;
+
+        if(tagType == 11){
+            rSAKc1[0] = sak & 0xFC & 0X70;
+            DbpString(_YELLOW_("[ ") "Passando no Sak Penegui" _YELLOW_(" ]"));
```
Comment (iceman1001) [inline armsrc/iso14443a.c:1194]: remove this debug statement.

File: armsrc/iso14443a.c [hunk 2]
```diff
@@ -1183,7 +1188,15 @@ bool SimulateIso14443aInit(uint8_t tagType, uint16_t flags, uint8_t *data, tag_r
...
+        }else{
+            rSAKc1[0] = sak & 0xFB;
+            //DbpString(_YELLOW_("[ ") "Passando no Sak Antigo" _YELLOW_(" ]"));
```
Comment (iceman1001) [inline armsrc/iso14443a.c:1197]: remove this one too

File: armsrc/Standalone/hf_msdsal.c [hunk 3]
```diff
@@ -185,11 +185,11 @@ void RunMod(void) {
     // in case there is a read command received we shouldn't break
     uint8_t data[PM3_CMD_DATA_SIZE] = {0x00};
 
-    uint8_t visauid[7] = {0x01, 0x02, 0x03, 0x04};
+    uint8_t visauid[7] = {0x05, 0x06, 0x07, 0x08};
```
Comment (iceman1001) [inline armsrc/Standalone/hf_msdsal.c:188]: Why this change?

File: armsrc/Standalone/hf_msdsal.c [hunk 4]
```diff
@@ -168,7 +168,7 @@ void RunMod(void) {
     //Expiration date: 17/11
     //Service code: 201
     //Discretionary data: 0000030000991
-    //char token[19] = {0x44,0x12,0x34,0x56,0x05,0x78,0x12,0x34,0xd1,0x71,0x12,0x01,0x00,0x00,0x03,0x00,0x00,0x99,0x1f};
+    //char token[19] = {0x44,0x12,0x34,0x56,0x05,0x78,0x56,0x78,0xd1,0x71,0x12,0x01,0x00,0x00,0x03,0x00,0x00,0x99,0x1f};
```
Comment (iceman1001) [inline armsrc/Standalone/hf_msdsal.c:171]: why this change ?

File: armsrc/iso14443a.c (hunk 2 above)
Comment (penegui) [reply armsrc/iso14443a.c:1197]: Sorry! I forgot to take it out of debug statement!

File: armsrc/Standalone/hf_msdsal.c (hunk 4 above)
Comment (penegui) [reply armsrc/Standalone/hf_msdsal.c:171]: It's not relevant, it will be removed.

File: armsrc/Standalone/hf_msdsal.c (hunk 3 above)
Comment (penegui) [reply armsrc/Standalone/hf_msdsal.c:188]: It's not relevant, it will be removed.

File: armsrc/Standalone/hf_msdsal.c [hunk 5]
```diff
@@ -386,30 +387,35 @@ void RunMod(void) {
...
 
                     // received a RATS request
                 } else if (receivedCmd[0] == ISO14443A_CMD_RATS && len == 4) {
-//                    DbpString(_YELLOW_("+") "Request for RATS");
+                    DbpString(_YELLOW_("+") "Request for RATS");
                     prevCmd = 0;
-                    p_response = &responses[RESP_INDEX_RATS];
+                    //p_response = &responses[RESP_INDEX_RATS];
+
+                    static uint8_t rRATS[] = { 0x13, 0x78, 0x80, 0x72, 0x02, 0x80, 0x31, 0x80, 0x66, 0xb1, 0x84, 0x0c, 0x01, 0x6e, 0x01, 0x83, 0x00, 0x90, 0x00 };
```
[the remark this answers is not in the document]
Comment (iceman1001) [reply armsrc/Standalone/hf_msdsal.c:415]: I looked at the original file.  Salva uses the dynamic quite a lot.  Its fine then.

**Comment by iceman1001** [comment]
Excellent,  thank you for your patience
